test/regression/external/submit-returns.spec.js

```diff
@@ -0,0 +1,440 @@
+    radioButton.click();
+    const button = getButton();
+    button.click();
+  });
+
+  it('How are you reporting your figures', () => {
+    expect($('form')).toBeVisible();
+    const radioLabel1 = getLabel('method');
+    expect(radioLabel1).toHaveTextContaining('Readings from a single meter');
+    const radioLabel2 = getLabel('method-2');
+    expect(radioLabel2).toHaveTextContaining('Volumes from one or more meters');
+    const radioLabel3 = getLabel('method-3');
+    expect(radioLabel3).toHaveTextContaining('Estimates without a meter');
+    const button = getButton();
+    button.click();
+  });
+
+  checkErrorMessage('Select readings from one meter, or other (abstraction volumes)');
+
+  it('selects Readings from a single meter', () => {
+    const radioButton = $('[value="oneMeter,measured"]');
+    radioButton.click();
+    const button = getButton();
+    button.click();
+  });
+
+  it('answers yes to Did your meter reset in this abstraction period?', () => {
+    expect($('form')).toBeVisible();
+    const formQuestion = $('//form/div/fieldset/legend');
+    expect(formQuestion).toHaveTextContaining('Did your meter reset in this abstraction period?');
+    const label1 = getLabel('meterReset');
+    expect(label1).toHaveTextContaining('Yes');
+    const label2 = getLabel('meterReset-2');
+    expect(label2).toHaveTextContaining('No');
+    const button = getButton();
+    button.click();
+  });
+
+  checkErrorMessage('Has your meter reset or rolled over?');
+
+  it('selects No to meter reset', () => {
+    const radioButton = $('[value="false"]');
+    radioButton.click();
+    const button = getButton();
+    button.click();
+  });
+
+  it('selects cubic meters as the unit measured', () => {
+    expect($('form')).toBeVisible();
+    const formQuestion = $('//form/div/fieldset/legend');
+    expect(formQuestion).toHaveTextContaining('Which units are you using?');
+    const label1 = getLabel('units');
+    expect(label1).toHaveTextContaining('Cubic metres');
+    const label2 = getLabel('units-2');
+    expect(label2).toHaveTextContaining('Litres');
+    const label3 = getLabel('units-3');
+    expect(label3).toHaveTextContaining('Megalitres');
+    const label4 = getLabel('units-4');
+    expect(label4).toHaveTextContaining('Gallons');
+    const button = getButton();
+    button.click();
+  });
+
+  checkErrorMessage('Select a unit of measurement');
+
+  it('selects cubic meters', () => {
+    const radioButton = $('[value="m³"]');
+    radioButton.click();
+    const button = getButton();
+    button.click();
+  });
+
+  it('enters the meter readings', () => {
+    expect($('form')).toBeVisible();
+    const formQuestion = $('//form/p');
+    expect(formQuestion).toHaveTextContaining('Enter your readings exactly as they appear on your meter');
+    const button = getButton();
+    button.click();
+  });
+
+  checkErrorMessage('Enter a meter start reading');
+
+  it('selects cubic meters', () => {
```

Comment:
There is additional logic here it would be good to add tests for:

* Initial reading cannot be negative
* Meter readings must not be negative
* Meter readings must keep increasing (e.g. March reading cannot be lower than Feb)

It would be good to include tests for:
* All fields left empty
* Some fields left empty
* Non-numeric values
* Zero values
* Negative values
* Volumes entered in each different unit available - are they calculated correctly

Reply:
Additional tests added as requested